gst.so/src/gst_worker.cpp

```diff
@@ -250,20 +241,11 @@ bool GSTWorker::do_gst_ramp(int *error, string *err_description) {
         //Start the timer
         start_time = gpu_blas->get_time_us();
 
-        // run GEMM & wait for completion
+        // run GEMM operation
         gpu_blas->run_blass_gemm(gst_ops_type);
 
-        /* Set callback to be called upon completion of blas gemm operations */
-        gpu_blas->set_callback(blas_callback, (void *)this);
-
-        std::unique_lock<std::mutex> lk(mutex);
-        cv.wait(lk);
-
-        if(!blas_status) {
-          msg = "[" + action_name + "] " + MODULE_NAME + " " +
-            std::to_string(gpu_id) + " " + " BLAS gemm operations failed !!! ";
-          rvs::lp::Log(msg, rvs::logtrace);
-        }
+        // Wait for GEMM operation to complete
+        gpu_blas->is_gemm_op_complete();
```

Comment:
this is immediate returning call, it might return false/true based on hipStreamQuery(hip_stream)  complete or not. we are not checking return here and incomplete gemm operation will still return above call. check in a loop for proper working

Reply:
is_gemm_op_complete() call is blocking now as we use hipStreamSynchronize() instead of hipStreamQuery().
Will add return checking for both run_blass_gemm() and is_gemm_op_complete().